src/cpp/session/modules/SessionTutorial.R

```diff
@@ -247,7 +247,18 @@
    
    tutorialFile <- tutorialFiles[[1]]
    contents <- readLines(tutorialFile, encoding = "UTF-8", warn = FALSE)
-   yaml <- rmarkdown:::parse_yaml_front_matter(contents)
+   
+   # find YAML header (if any). note that we avoid using rmarkdown here
+   # just because we want to avoid 
+   reYamlSeparator <- "^\\s*(?:---|[.][.][.])\\s*$"
```

Comment:
Is this a common regex, i.e. we use it elsewhere for this purpose? Don't know that we have any way to share such things in our embedded R code, mainly curious if this is tried and tested vs. something new.

Reply:
We use it for highlight rules here:

https://github.com/rstudio/rstudio/blob/3e709a0b099ee3894d4b2cd85424ae2eaf5c3e54/src/gwt/acesupport/acemode/rmarkdown_highlight_rules.js#L156-L164

And we have a similar version here:

https://github.com/rstudio/rstudio/blob/3e709a0b099ee3894d4b2cd85424ae2eaf5c3e54/src/cpp/core/YamlUtil.cpp#L32-L36

The Pandoc [spec](https://pandoc.org/MANUAL.html#extension-yaml_metadata_block) says:

> A YAML metadata block is a valid YAML object, delimited by a line of three hyphens (---) at the top and a line of three hyphens (---) or three dots (...) at the bottom.

So, strictly speaking, the second version is wrong, and so is the version in this PR -- although in practice, users almost always use `---`.